utils/simplecache/cache.go

```diff
@@ -0,0 +1,210 @@
+package simplecache
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/ARM-software/golang-utils/utils/commonerrors"
+	fs "github.com/ARM-software/golang-utils/utils/filesystem"
+	"github.com/ARM-software/golang-utils/utils/parallelisation"
+)
+
+const (
+	_DefaultGCPeriod = 10 * time.Minute
+	_DefaultTTL      = 2 * time.Hour
+)
+
+type CacheEntry struct {
+	path         string
+	originalFS   fs.FS
+	originalpath string
+	expiration   time.Time
```

Comment:
the source may not be a file in a filesystem but something downloaded from the internet
so there should be a func() to retrieve entry